nifi-nar-bundles/nifi-adx-bundle/nifi-adx/src/main/java/org/apache/nifi/adx/NiFiVersion.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.adx;
+
+import org.apache.commons.lang3.tuple.Pair;import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.util.Properties;
+
+public class NiFiVersion {
+    public static final String CLIENT_NAME = "Kusto.Nifi";
+
+    public static final Pair<String,String> NIFI_SOURCE = Pair.of("processor", "nifi-source");
+    private static final Logger log = LoggerFactory.getLogger(NiFiVersion.class);
+    private static final String VERSION_FILE = "/azure-kusto-nifi-version.properties";
+    private static String version = "unknown";
+
+    private NiFiVersion() {
+    }
+
+    static {
+        try {
+            Properties props = new Properties();
+            try (InputStream versionFileStream = NiFiVersion.class.getResourceAsStream(VERSION_FILE)) {
+                props.load(versionFileStream);
+                version = props.getProperty("version", version).trim();
+            }
+        } catch (Exception e) {
+            log.warn("Error while loading version:", e);
+        }
```

Comment:
This approach should be refactored to use `getPackage().getImplementationVersion()` along with adding the information to the module manifest configuration.